lib/api/unittest/CJsonOutputWriterTest.cc

```diff
@@ -1056,6 +1056,110 @@ BOOST_AUTO_TEST_CASE(testSimpleWrite) {
+    {
+        ml::core::CJsonOutputStreamWrapper outputStream(sstream);
+        ml::api::CJsonOutputWriter writer("job", outputStream);
+        TDouble1Vec actual(2, 0.0);
+        actual[0] = 40.0;
+        actual[1] = -40.0;
+        TDouble1Vec typical(2, 0.0);
+        typical[0] = 90.0;
+        typical[1] = -90.0;
+        ml::api::CHierarchicalResultsWriter::SResults result(
+                false, false, partitionFieldName, partitionFieldValue,
+                overFieldName, overFieldValue, byFieldName, byFieldValue,
+                correlatedByFieldValue, 1, function, functionDescription,
+                typical, actual, 2.24, 0.5, 10.0,
+                79, fieldName, influences, false, true, 1, 100);
+        writer.acceptResult(result);
+        BOOST_TEST_REQUIRE(writer.acceptResult(result));
+        BOOST_TEST_REQUIRE(writer.endOutputBatch(true, 1U));
+    }
+
+    {
+        ml::core::CJsonOutputStreamWrapper outputStream(sstream);
+        ml::api::CJsonOutputWriter writer("job", outputStream);
+        TDouble1Vec actual(1, 500);
+        TDouble1Vec typical(1, 64);
+        ml::api::CHierarchicalResultsWriter::SResults result(
+                false, false, partitionFieldName, partitionFieldValue,
+                overFieldName, overFieldValue, byFieldName, byFieldValue,
+                correlatedByFieldValue, 1, function, functionDescription,
+                typical, actual, 2.24, 0.5, 10.0,
+                79, fieldName, influences, false, true, 1, 100);
+        BOOST_TEST_REQUIRE(writer.acceptResult(result));
+        BOOST_TEST_REQUIRE(writer.endOutputBatch(true, 1U));
+    }
+
+    {
+        ml::core::CJsonOutputStreamWrapper outputStream(sstream);
+        ml::api::CJsonOutputWriter writer("job", outputStream);
+        TDouble1Vec actual(2, 0.0);
+        actual[0] = 40.0;
+        actual[1] = -40.0;
+        TDouble1Vec typical(2, 0.0);
+        typical[0] = 90.0;
+        typical[1] = -90.0;
+        ml::api::CHierarchicalResultsWriter::SResults result(
+                false, false, partitionFieldName, partitionFieldValue,
+                overFieldName, overFieldValue, byFieldName, byFieldValue,
+                correlatedByFieldValue, 1, "mean", functionDescription,
+                typical, actual, 2.24, 0.5, 10.0,
+                79, fieldName, influences, false, true, 1, 100);
+        BOOST_TEST_REQUIRE(writer.acceptResult(result));
+        BOOST_TEST_REQUIRE(writer.endOutputBatch(true, 1U));
+    }
+
+    LOG_WARN(<< sstream.str());
+    rapidjson::Document arrayDoc;
+    arrayDoc.Parse<rapidjson::kParseDefaultFlags>(sstream.str().c_str());
+
+    BOOST_TEST_REQUIRE(arrayDoc.IsArray());
+    BOOST_REQUIRE_EQUAL(rapidjson::SizeType(3), arrayDoc.Size());
+
+    const rapidjson::Value& object = arrayDoc[rapidjson::SizeType(0)];
+    BOOST_TEST_REQUIRE(object.IsObject());
+
+    BOOST_TEST_REQUIRE(object.HasMember("typical"));
+    //BOOST_REQUIRE_EQUAL(2, object["typical"].GetArray().Size());
```

Comment:
Hmmm.. I'm always in 2 minds about commented out code, maybe delete it if it's not needed?

Reply:
yeah, I cannot get the tests to do what I want the to do :(. So, very WIP there.